### ISSUE TITLE
How to chat with history?

### ISSUE THREAD

**Issue by redthing1** [nomic-ai/pyllamacpp#37]
I want to use this to run a chat session so that future generate calls have the previous context. But I don't want to re-tokenize and re-run the old tokens again right? Is there some easy way to do this?

**Comment by abdeladim-s** [comment]
Yes you can use interactive mode in the generate function, it behaves the same as the `main` example of `llama.cpp`, I will add a CLI soon to see how it works. 
I will make it more convenient soon!

**Comment by mattorp** [comment]
What's the approach for this when using the langchain wrapper?

https://github.com/hwchase17/langchain/blob/master/langchain/llms/gpt4all.py

**Comment by liftoff1968** [comment]
A way of accessing the history of a conversation would be great. Just like GPT-4.
 messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Who won the world series in 2020?"},
        {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."},
        {"role": "user", "content": "Where was it played?"}
    ]

**Comment by abdeladim-s** [comment]
Hi @redthing1,

The history now is kept as long as the model variable is alive (see Please try [Interactive Dialogue](https://github.com/abdeladim-s/pyllamacpp/tree/main#interactive-dialogue) from the readme page.)

You can call `model.reset()` to reset the history.
